crates/connectors/ndc-postgres/src/state.rs

```diff
@@ -18,14 +18,16 @@ use query_engine_execution::metrics;
 pub struct State {
     pub pool: PgPool,
     pub database_info: DatabaseInfo,
-    pub metrics: metrics::Metrics,
+    pub query_metrics: metrics::Metrics,
+    pub configuration_metrics: ndc_postgres_configuration::Metrics,
```

Comment:
I would prefer a single `metrics` field with a `Metrics` type, and that type could be a struct which is composed from both types.
I would even prefer all metrics to be handled in one place, but would be fine with an abstraction :)